common/app/views/fragments/img.scala.html

```diff
@@ -1,74 +1,109 @@
+        itemscope
+        itemtype="http://schema.org/ImageObject"
+        data-component="image"
+        class="@figureClasses @shareInfo.map { _ => fig--narrow-caption fig--has-shares} @if(!imageOption.map(_.showCaption).getOrElse(false)){fig--no-caption}"
+        @defining(imageOption.flatMap(_.mediaId)){ id =>
+            @id.map{ id => data-media-id="@id" }
+        }
+        @lightboxIndex.map { index =>
+          id="img-@index"
+        }
+    >
+
+        @if(isMain) {
+            <meta itemprop="representativeOfPage" content="true">
+        }
+
+        @lightbox{
+            @image(picture, widthsByBreakpoint, imageOption, isMain)
+        }
+        @showcaseFeature{
+            @caption(picture, imageOption, isMain)
         }
     </figure>
+
 }
 
-@lightbox(picture: model.ImageContainer, widths: WidthsByBreakpoint) =  {
+@showcaseFeature(captionHtml: => Html) =  {
 
-    @if(isLightboxable) {
-        <a href="@url#img-1" class="article__img-container js-gallerythumbs" data-link-name="Launch Article Lightbox" data-is-ajax>
-            <div class="@if(isFeature && isShowcase){u-responsive-ratio u-responsive-ratio--letterbox}">
-                @image(picture, widths)
-                @fragments.inlineSvg("expand-image", "icon", List("centered-icon rounded-icon article__fullscreen"))
+    @if(isFeature && isShowcase) {
+        <div class="gs-container">
+            <div class="content__main-column">
+                @captionHtml
             </div>
-        </a>
+        </div>
     } else {
-        @image(picture, widths)
+        @captionHtml
     }
 
 }
 
-@image(picture: model.ImageContainer, widths: WidthsByBreakpoint) = {
+@lightbox(imageHtml: => Html) =  {
 
-    <img src="@ImgSrc.getFallbackUrl(picture)"
-        srcset="@ImgSrc.srcset(picture, widths)"
-        sizes="@widths.sizes"
-        class="maxed responsive-img"
-        itemprop="contentUrl"
-        alt="@ImgSrc.getFallbackAsset(picture).flatMap(_.altText).getOrElse("")" />
+    @lightboxIndex.map { index =>
+        <a href="#img-@index" class="article__img-container js-gallerythumbs" data-link-name="Launch Article Lightbox" data-is-ajax>
+                @imageHtml
+                @fragments.inlineSvg("expand-image", "icon", List("centered-icon rounded-icon article__fullscreen"))
+        </a>
+        @shareInfo.map { case (shareLinks, contentType) =>
+            @share.blockLevelSharing(s"img-$index", shareLinks, contentType)
+        }
+    }.getOrElse {
+        @imageHtml
+    }
+
+}
+
+@image(picture: model.ImageContainer, widths: WidthsByBreakpoint, imageOption: Option[ImageAsset], isMain: Boolean) = {
+
+    @picture.largestImage.map{ largestImage =>
+        <div
+            class="u-responsive-ratio@if(isFeature && isShowcase) { u-responsive-ratio--letterbox}"
+            @if(isFeature && isShowcase) {} else {style="padding-bottom: @{"%.2f".format(100 * largestImage.height.toDouble / largestImage.width)}%"}
```

Comment:
although style= generally indicates mixing content and layout, in this case in reality it's metadata about content, it's not actually styling anything in reality.  Think of it like img width= and height=, only it's defining an aspect ratio instead of absolute sizes.
If I wanted to use CSS for this, I can't see any way of avoiding having an excessive number of classes, one for each possible ratio.